### PR TITLE
feat(component): add form-errors custom element

### PR DESCRIPTION
**Pull request by bas080**
Add a globalized form-errors component. Also adds the association-select's type bindable.

### CODE REVIEW

File: src/component/form-errors.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<template>
+  <p class="help-block" repeat.for="message of messages">${message}</p>
+</template>
+
```
Comment (RWOverdijk) [inline src/component/form-errors.html]: Remove newline.

File: src/component/framework/bootstrap/form-errors.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<template>
+  <p repeat.for="message of messages">${message}</p>
+</template>
+
```
Comment (RWOverdijk) [inline src/component/framework/bootstrap/form-errors.html]: Remove newline.

File: src/config.js [hunk 3]
```diff
@@ -8,7 +8,13 @@ export class Config {
...
       labelFormat: element => {
         return element.label || element.key;
-      }
+      },
+
+      // By default is configured to work with aurelia validation
+      // ValidationResult instance
+      errorMessage: error => error.message,
+      errorValid  : error => error.valid
+
```
Comment (RWOverdijk) [inline src/config.js]: Remove newline

File: src/component/framework/form-group.js [hunk 4]
```diff
@@ -10,10 +10,22 @@ export class FormGroup {
 
   @bindable description
 
+  @bindable errors = []
+
   constructor(config) {
+    this.config = config.configurations;
```
Comment (RWOverdijk) [inline src/component/framework/form-group.js]: Align consecutive assignments

File: src/component/framework/bootstrap/association.html [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
+  <form-group errors.bind="errors" role="group" element.bind="formField.element" message.bind="formField.message" description.bind="formField.description">
     <association-select
       attributes.bind        = "formField.element.attributes"
       multiple.bind          = "formField.element.multiple"
       value.bind             = "formField.value"
       property.bind          = "formField.element.property"
       repository.bind        = "formField.element.repository"
       resource.bind          = "formField.element.resource"
-
+      type.bind              = "formField.element.selectType"
```
Comment (RWOverdijk) [inline src/component/framework/bootstrap/association.html:12]: What's this?

File: src/component/form-fields.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ export class FormFields {
   @bindable({defaultBindingMode: bindingMode.twoWay})
   model
 
-  @bindable messages
+  @bindable errors
```
Comment (RWOverdijk) [inline src/component/form-fields.js]: Semicolon (apply everywhere);

File: src/component/form-errors.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    return this.errors ? this.errors.filter(error => !this.config.errorValid(error)) : [];
+  }
+
+  @computedFrom('errors')
+  get messages() {
+    return this.invalidErrors.map(this.config.errorMessage);
+  }
+
+}
+
```
Comment (RWOverdijk) [inline src/component/form-errors.js]: Remove newline

File: src/component/form-errors.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import {inject, customElement, bindable, containerless, computedFrom} from 'aurelia-framework';
+import {Config} from '../config';
+
+@customElement('form-errors')
+@containerless
+@inject(Config)
+export class FormErrors {
+
+  @bindable
+  errors;
```
Comment (RWOverdijk) [inline src/component/form-errors.js]: Put on same line as @bindable

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/component/framework/bootstrap/association.html (hunk 5 above)
Comment (bas080) [reply src/component/framework/bootstrap/association.html:12]: This allows for checkboxes or select type to be chose for aurelia-orm's association-select component.

Comment (bas080) [reply src/component/framework/bootstrap/association.html:12]: It might be best to make this part of a seperate pull request

**Pull request closed without merging**